@here/harp-datasource-protocol/lib/InterpolatedProperty.ts

```diff
@@ -344,12 +348,16 @@ function procesStringEnocodedNumeralInterpolatedProperty(
                 needsMask = true;
             }
 
-            const result = valueFormat.decoder(value);
-            for (let i = 0; i < result.length; ++i) {
-                propValues[valueIdx * valueFormat.size + i] = result[i];
+            for (let i = 0; i < valueFormat.size; ++i) {
+                propValues[valueIdx * valueFormat.size + i] = tmpBuffer[i];
             }
             break;
         }
+        if (!matched) {
+            throw Error(
+                `Not all interpolation values match the same format: ${JSON.stringify(prop)}`
+            );
+        }
```

Comment:
@robertoraggi I think this was missing entirely (even in the old flow/logic), because you could create interpolated property like this:
`
 "color": {
                        "interpolation": "Linear",
                        "zoomLevels": [0, 11, 12, 15, 16, 20],
                        "values": ["#f00", "2m", "#f00", "#f00", "#f00", "#f00"]
                    }
`

>